File: cpp/kode program/Electronic.cpp
#include <iostream>
#include "Product.cpp"
using namespace std;

// class electronic adalah turunan dari class product
class Electronic : public Product {
    private:
        string brand;     // atribut brand untuk menyimpan merek
        string warranty;  // atribut warranty untuk menyimpan garansi
        string category;  // atribut category untuk menyimpan kategori produk

    public:
    // constructor kosong
    Electronic() {

    }

    // constructor dengan parameter yang memanggil constructor dari product
    Electronic(int id, string name, int price, int stock, string image,
                string brand, string warranty, string category) : Product(id, name, price, stock, image){
                this->brand = brand;
                this->warranty = warranty;
                this->category = category;
            }

    // setter dan getter untuk brand
    void setBrand(string brand) {
        this->brand = brand;
    }
    string getBrand() {
        return this->brand;
    }

    // setter dan getter untuk warranty
    void setWarranty(string warranty) {
        this->warranty = warranty;
    }
    string getWarranty() {
        return this->warranty;
    }

    // setter dan getter untuk category
    void setCategory(string category) {
        this->category = category;
    }
    string getCategory() {
        return this->category;
    }
};

File: php/kode program/Electronic.php
<?php
require_once "Product.php"; // mengimpor class Product agar bisa digunakan sebagai parent

// class Electronic mewarisi (extends) dari Product
class Electronic extends Product {
    // atribut khusus Electronic
    private $brand;     // merek barang
    private $warranty;  // masa garansi
    private $category;  // kategori barang (misal: Laptop, TV, dll)

    // constructor untuk inisialisasi objek Electronic
    public function __construct($id, $name, $price, $stock, $image, $brand, $warranty, $category) {
        // memanggil constructor parent (Product) agar atribut dasar ikut terisi
        parent::__construct($id, $name, $price, $stock, $image);
        // mengisi atribut tambahan milik Electronic
        $this->brand = $brand;
        $this->warranty = $warranty;
        $this->category = $category;
    }

    // getter
    // mengambil nilai brand
    public function getBrand() { return $this->brand; }
    // mengambil nilai warranty
    public function getWarranty() { return $this->warranty; }
    // mengambil nilai category
    public function getCategory() { return $this->category; }

    // setter
    // mengubah nilai brand
    public function setBrand($brand) { $this->brand = $brand; }
    // mengubah nilai warranty
    public function setWarranty($warranty) { $this->warranty = $warranty; }
    // mengubah nilai category
    public function setCategory($category) { $this->category = $category; }
}
?>
